airmozilla/main/views.py

```diff
@@ -482,6 +485,45 @@ def get(self, request, slug):
 
         return render(request, self.template_name, context)
 
+
+def es_to_template(self, request, slug):
+    es = pyelasticsearch.ElasticSearch('http://localhost:9200/')
+    for event in Event.objects.scheduled().order_by('?')[:100]:
+        # should do bulk ops really but besides the point
+        es.index(
+            'events',
+            'event',
+            {
+                'title': event.title,
+                'tags': [x.name for x in event.tags.all()],
+                'channels': [x.name for x in event.channels.all()],
+            },
+            id=event.id,
+        )
+
+    es.refresh()  # let it clear its throat
+    print '-' * 100
+
+    # get one my id
+    event, = Event.objects.scheduled().order_by('?')[:1]
+    try:
+        doc = es.get('events', 'event', event.id)
+        pprint(doc['_source'])
+        assert doc['_source']['title'] == event.title
+    except pyelasticsearch.exceptions.ElasticHttpNotFoundError:
+        print "Not indexed yet :("
+
+    print '-' * 100
+    hits = es.search('title: firefox', index='events')['hits']
+    for doc in hits['hits']:
+        event = Event.objects.get(id=doc['_id'])
+        print "\t", repr(event.title), doc['_score']
+   
+    template_name = 'main/es.html'
+    ids = [int(x) for x in id.split(',')]
+    events = Event.objects.filter(id__in=ids)
```

Comment:
So, the order might be something that's driven from ES. Basically, once you've figured out how to tell ES to score the results and have them sorted by that, you'll want to preserve the order that ES returned them in. One way to do that would be like this:
```python
hits = es.search('title: firefox', index='events')['hits']
ids = []
for doc in hits['hits']:
    ids.append(doc['_id'])
events = Event.objects.filter(id__in=ids)
events.sort(lambda e: ids.index(e.id))
```
That way, you get them in the order that ES had them. And you're only doing 1 django ORM query. 

That, would get you going. There's going to be a lot of necessary optimization but that's something that can wait.